authors/apps/articles/views.py

```diff
@@ -45,29 +49,223 @@ class RetrieveUpdateDeleteArticleAPIView(RetrieveUpdateAPIView):
+
+    def get(self, request, slug):
+        """
+        Method to return ratings of a particular article
+
+        Params
+        -------
+        request: Object with request data and functions.
+        slug: reference to article to be returned
+
+        Returns
+        --------
+        ratings
+        error message if not found
+        
+        """
+        ratings = Ratings.objects.all()
+        serializer = RatingsSerializer(ratings, many=True)
+        filtered_ratings = [x for x in serializer.data if x['slug']==slug]
+        if filtered_ratings:
+            return Response({'data': filtered_ratings, 'RatingsCount': len(serializer.data)}, status=status.HTTP_200_OK)
+        return Response({'errors': 'no ratings for this article present'}, status=status.HTTP_404_NOT_FOUND)
+
+    def post(self, request, slug):
+        """
+        Method to post a rating of a particular article
+
+        Params
+        -------
+        request: Object with request data and functions.
+        slug: reference to article to be returned
+
+        Returns
+        --------
+        rating that has just been posted
+        error message if:
+                - you try to rate your own article
+                - you try to rate an article twice
+                - any other validation errors encountered
+        
+        """
+        rating = request.data.get('rating', {})
+        serializer = self.serializer_class(data=rating)
+        serializer.is_valid(raise_exception=True)
+
+        article = self.get_object(slug)
+        
+        article_id = article.id
+        author = self.request.user
+
+        if article.author == author:
+            return Response({'errors': 'cannot rate own article'}, status=status.HTTP_400_BAD_REQUEST)
+
+        try:
+            Ratings.objects.get(article_id=article_id, author=author)
+            return Response({'errors': 'cannot rate an article twice'}, status=status.HTTP_400_BAD_REQUEST)
+        except:
+            serializer.save(author=request.user, article=article)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        
+        
+class RetrieveUpdateDeleteRatingAPIView(APIView):
+    """
+    Allow only authenticated users to hit these endpoints.
+    List edit or delete a rating
+    only a rating owner can edit or delete his/her article
+    """
+    permission_classes = (IsAuthenticatedOrReadOnly, IsOwnerOrReadOnly, IsVerifiedUser )
+    serializer_class = RatingsSerializer
+    renderer_classes = (RatingJSONRenderer,)
+
+    def get_object(self, pk):
+        """
+        Method to return an article 
+
+        Params
+        -------
+        slug: referende to article to be returned
+
+        Returns
+        --------
+        an article object if found
+        raises an excepition if not found
+        
+        """
+        try:
+            return Ratings.objects.get(pk=pk)
+        except Ratings.DoesNotExist:
+            raise ArticleNotFound
```

Comment:
This should raise a `Rating not found` error message as opposed to `Article not found`